src/eval_context.rs

```diff
@@ -893,6 +893,10 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         }
         // FIXME: assuming here that type size is < i64::max_value()
         let pointee_size = self.type_size(pointee_ty)?.expect("cannot offset a pointer to an unsized type") as i64;
+        if pointee_size == 0 {
+            // rustc relies on offsetting pointers to zsts to be a nop
```

Comment:
Where does this occur? So far I've only observed "offset-by-0" on ZSTs. If the offset is >0, this sounds very fishy from an LLVM perspective.